### ISSUE TITLE
v5.0.6 Doesn't Register 'bigInteger' Column Type with Doctrine?

### ISSUE THREAD

**Issue by jakecarlson** [laravel/framework#7560]
After updating to v5.0.6, I'm getting the following error when running migrations which reference 'bigInteger' columns. v5.0.5 does not seem to have this issue.

[Doctrine\DBAL\DBALException]                                                                                        
  Unknown column type "bigInteger" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL  
  \Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If  
   this error occurs during database introspection then you might have forgot to register all database types for a Do  
  ctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedD  
  atabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping informatio  
  n.

**Comment by GrahamCampbell** [comment]
Likely you're using an unsupported version of doctrine.

**Comment by GrahamCampbell** [comment]
Nothing has change in our code between those two versions.

**Comment by GrahamCampbell** [comment]
The db component is actually totally identical: https://github.com/illuminate/database/compare/v5.0.5...v5.0.6.

**Comment by jakecarlson** [comment]
I've honed in on the specific problem: running the change() method on an existing column, e.g.: 

Schema::table('test', function(Blueprint $table)
        {
            $table->unsignedBigInteger('test')->nullable()->default(null)->change();
        }
);

I've tried with the most recent versions of dbal 2.3, 2.4, and 2.5. In Laravel 5.0.5, it works in all 3. In Laravel 5.0.6, it fails in all 3. It's almost like the change() method specifically doesn't translate the field mapping properly.

As an aside, the L5 docs mention using doctrine/dbal for extended functionality for migrations, but so far as I could see they do not specify *which* versions are compatible.

**Comment by GrahamCampbell** [comment]
@jakecarlson Can you identify which commit broke this please: https://github.com/laravel/framework/commits/5.0/src/Illuminate/Database.

**Comment by jakecarlson** [comment]
Looks like this broke it: https://github.com/laravel/framework/commit/72d3febb752fb252148f8cd7a677587db8f20d04

**Comment by GrahamCampbell** [comment]
@jakecarlson Could you prepare a fix that doesn't break the other issue that required that change please?

**Comment by jakecarlson** [comment]
Ok I created a pull request: https://github.com/laravel/framework/pull/7562 . First time contributor so let me know if I did it the right way. I made the change to have minimal impact and only affect the change method (I think), but there may be a more elegant solution that can be made globally to type mapping.

**Comment by rilut** [comment]
I've got a similar error too for my migration

My migraton:
```
public function up()
{
	Schema::table('statistics', function(Blueprint $table)
	{			
		$table->dateTime('bestProductiveTime')->nullable()->change();
		$table->dateTime('worstProductiveTime')->nullable()->change();
	});
}
```

The error:
```
[Doctrine\DBAL\DBALException]
Unknown column type "dateTime" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgot to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.
```

From http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/basic-mapping.html, the type's should be `datetime` rather than `dateTime` Could it affects all type?

**Comment by GrahamCampbell** [comment]
And what does `getTypesMap()` give?

**Comment by rilut** [comment]
```
{
array: "Doctrine\DBAL\Types\ArrayType",
simple_array: "Doctrine\DBAL\Types\SimpleArrayType",
json_array: "Doctrine\DBAL\Types\JsonArrayType",
object: "Doctrine\DBAL\Types\ObjectType",
boolean: "Doctrine\DBAL\Types\BooleanType",
integer: "Doctrine\DBAL\Types\IntegerType",
smallint: "Doctrine\DBAL\Types\SmallIntType",
bigint: "Doctrine\DBAL\Types\BigIntType",
string: "Doctrine\DBAL\Types\StringType",
text: "Doctrine\DBAL\Types\TextType",
datetime: "Doctrine\DBAL\Types\DateTimeType",
datetimetz: "Doctrine\DBAL\Types\DateTimeTzType",
date: "Doctrine\DBAL\Types\DateType",
time: "Doctrine\DBAL\Types\TimeType",
decimal: "Doctrine\DBAL\Types\DecimalType",
float: "Doctrine\DBAL\Types\FloatType",
binary: "Doctrine\DBAL\Types\BinaryType",
blob: "Doctrine\DBAL\Types\BlobType",
guid: "Doctrine\DBAL\Types\GuidType"
}
```

**Comment by GrahamCampbell** [comment]
Looks like we need to just use all lower case there then?

**Comment by GrahamCampbell** [comment]
I can write a fix if you want?

**Comment by GrahamCampbell** [comment]
Which version of doctrine is this btw?

**Comment by GrahamCampbell** [comment]
How does #8092 look?

**Comment by rilut** [comment]
@GrahamCampbell It works, I've applied your #8092 for a `dateTime` column (yes, `datetime` in doctrine)
Great job!

**Comment by jakecarlson** [comment]
Yes this patch fixes it for my bigint issue as well on 5.0.18.

**Comment by ashishsanjayrao** [comment]
The TypeMap doesn't include the tinyint type too. Could that be fixed?
